custom_model_runner/datarobot_drum/drum/utils.py

```diff
@@ -230,8 +230,11 @@ def read_structured_input_data_as_df(binary_data, mimetype):
         try:
             if mimetype == PredictionServerMimetypes.TEXT_MTX:
                 return pd.DataFrame.sparse.from_spmatrix(mmread(io.BytesIO(binary_data)))
-            elif mimetype == PredictionServerMimetypes.APPLICATION_X_APACHE_ARROW_STREAM:
-                df = pyarrow.ipc.deserialize_pandas(binary_data)
+            elif (
+                mimetype == PredictionServerMimetypes.APPLICATION_X_APACHE_ARROW_STREAM
+                and get_pyarrow_module() is not None
```

Comment:
I think we should check mush earlier if format is supported. I'll leave general comment.